ecleankernel/__main__.py

```diff
@@ -256,6 +264,40 @@ def main(argv: typing.List[str]) -> int:
                 bootloader=bootloader,
                 destructive=args.destructive)
 
+            if hook_dirs:
+                preserved_kernels = [k for k in kernels
+                                     if k not in removals]
+
+                def as_version_list(kernels: typing.Iterable[Kernel]) -> str:
+                    versions = [x.version for x in kernels]
+                    return " ".join(versions)
+
+                env = os.environ.copy()
+
+                preserved_str = as_version_list(preserved_kernels)
+                env["ECLEAN_KERNEL_PRESERVED"] = preserved_str
+
+                removal_str = as_version_list(removals.keys())
+                env["ECLEAN_KERNEL_REMOVAL_CANDIDATES"] = removal_str
+
+                all_str = as_version_list(kernels)
+                env["ECLEAN_KERNEL_ALL"] = all_str
+
+                for hook_dir in hook_dirs:
+                    for hook in hook_dir.iterdir():
+                        if not hook.is_file():
+                            continue
+
+                        # Skip dotfiles.
+                        if hook.name[0] == '.':
```

Comment:
`""` per PEP 8. Also, I'd go with `.startswith()`, even though `name` couldn't be empty.